### PR TITLE
Adding magnet class

### LINKED ISSUE #1: There is no code
Nothing at all can be done since there is no code whatsoever

### PR DESCRIPTION
**Pull request by olimasters**
A bit of code added - namely, a magnet class, a makefile, and a main file (which does very little).  Resolves #1

### CODE REVIEW

File: Magnet.cpp [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+        for(int j=0;j<width;j++)
+            spins[i].push_back(distribution(gen));
+    }
+}
+
+bool Magnet::update(int i,int j)
+{
+    if(updatePending(i,j))
+    {
+        spins[i][j] = spins[i][j] == 1 ? -1 : 1;
```
Comment (AlexSWall) [inline Magnet.cpp]: What are the possible values of spins[i][j]? I believed it to just be {1,-1} always.

Comment (olimasters) [reply Magnet.cpp]: You are correct, spins[i][j] has to be either 1 or -1 for any i, j.

Comment (AlexSWall) [reply Magnet.cpp]: Then isn't this code [edit:] not optimum? What is wrong with spins[i][j] = -spins[i][j]?

Comment (olimasters) [reply Magnet.cpp]: Yep, I'd agree with your pre-edit post - it's vile.  I think my silly premature optimisation-obsessed brain was thinking that checking equality to 1 then setting to -1 or 1 might result in fewer instructions than multiplying by -1, but it's negligible compared to the other stuff we're doing anyway (like generating random numbers) if it's even better at all.  Changed accordingly.

**Comment by olimasters** [comment]
I'll merge this then if you have no more comments, @AlexSWall

**Review by ctlclarke — APPROVED** [review]

**Pull request merged**